src/main/java/com/wavefront/sdk/jersey/WavefrontJerseyFilter.java

```diff
@@ -63,6 +108,29 @@ public void filter(ContainerRequestContext containerRequestContext) throws IOExc
       String finalClassName = pair._1;
       String finalMethodName = pair._2;
 
+      if (tracer != null) {
+        String finalMatchingPath = "";
+        Resource matchedResource = request.getUriInfo().getMatchedModelResource();
+        if (matchedResource != null) {
+          finalMatchingPath = stripLeadingAndTrailingSlashes(matchedResource.getPath());
+          while (matchedResource.getParent() != null) {
```

Comment:
I think this is slightly inefficient since we invoke similar logic in 
Optional<String> optional = MetricNameUtils.metricName(request);

Please refactor MetricNameUtils.metricName to return Optional<Pair<JerseyPath>>
and make this method
private static Optional<String> metricName(String httpMethod, String path) {
package public.
That way you calculate JerseyPath only once.